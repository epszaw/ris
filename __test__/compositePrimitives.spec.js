const { schema, validate } = require('src/index')

describe('composition with default value', () => {
  it('should replace failed', () => {
    const primitivesSchema = {
      foo: [schema.string, 'bar'],
    }

    expect(
      validate(primitivesSchema, {
        foo: null,
      }),
    ).toEqual({
      foo: 'bar',
    })
  })

  it('should replace multiple failed', () => {
    const primitivesSchema = {
      foo: [schema.string, 'bar'],
      beep: [schema.number, 1],
    }

    expect(
      validate(primitivesSchema, {
        foo: null,
        beep: undefined,
      }),
    ).toEqual({
      foo: 'bar',
      beep: 1,
    })
  })
})

describe('composition with meta rules', () => {
  it('should correctly validate max and min numbers', () => {
    const compositeSchema = {
      foo: [schema.number, schema.min(2), schema.max(5)],
    }

    expect(
      validate(compositeSchema, {
        foo: 3,
      }),
    ).toEqual({
      foo: 3,
    })
    expect(() =>
      validate(compositeSchema, {
        foo: 8,
      }),
    ).toThrow()
    expect(() =>
      validate(compositeSchema, {
        foo: 1,
      }),
    ).toThrowErrorMatchingSnapshot()
  })
})
