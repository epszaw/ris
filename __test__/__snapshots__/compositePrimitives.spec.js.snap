// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`validate should throw error with reason if data not match to scheme 1`] = `
"Validation error: boop expected Boolean got foo
Validation error: beep expected null got 1"
`;
